### ISSUE TITLE
Shows all the months with data 0

### ISSUE THREAD

**Issue by GiovanniFrigo** [asdf1899/andamento-mensile#1]
Hey! 

Nice component ;) 

Unfortunately, it does not seem to work properly on my machine. This is what I see on the page:

<img width="1161" alt="image" src="https://user-images.githubusercontent.com/7932522/96690835-77b69f80-1384-11eb-9e1d-4b7b8a087db3.png">

The component is interactive (I can select and de-select the single months), but unfortunately all the months show 0 as value.
Tested on Firefox, Safari, Chrome.

**Comment by asdf1899** [comment]
Hi @GiovanniFrigo!

First of all I wanted to apologize whether the code isn't clean and well documented but yesterday I ran out of time to complete the component.

All the month shows 0 as value probably because the component doesn't load "data.json" file in "src/data" folder.

Are there any errors on the browser console?

I still apologize for the inconvenience.

**Comment by GiovanniFrigo** [comment]
The data is loaded correctly and no error are shown on the console, I just get a warning (`./src/components/Table.js
  Line 34:25:  Expected to return a value in arrow function  array-callback-return`) but that's not linked nor relevant.

In fact, if I try to log the data from the component (by adding a `console.dir(this.props.data);` to the componentDidMount), it works.

I'm testing on MacOS, but also happens on my windows PC

**Comment by asdf1899** [comment]
What does it print, if you try log "data" argument in filterData(data) function in "src/components/Table.js" file line 16?
It should print something like this:
```
0: {documents: 4, revenue: 15000, month: "2019/01"}
1: {documents: 4, revenue: 22341, month: "2019/02"}
2: {documents: 5, revenue: 32000, month: "2019/03"}
3: {documents: 1, revenue: 900, month: "2019/04"}
4: {documents: 2, revenue: 29000, month: "2019/05"}
5: {documents: 1, revenue: 28000, month: "2019/06"}
6: {documents: 2, revenue: 28000, month: "2019/07"}
7: {documents: 1, revenue: 12000, month: "2019/08"}
8: {documents: 0, revenue: 0, month: "2019/09"}
9: {documents: 3, revenue: 32000, month: "2019/010"}
10: {documents: 3, revenue: 32000, month: "2019/11"}
11: {documents: 2, revenue: 28000, month: "2019/12"}

```

In short words, filterData(data) function check in each "data" variable row whether "month" value is a valid date otherwise it keep "mesi" array params to 0({documents: 0, revenue: 0, month: 0}). 
(I should rewrite this function in a better way)

filterData(data) return "mesi" array and it should be something like this:

```
0: {0: "Gennaio", 1: {documents: 4, revenue: 15000, month: "2019/01"}}
1: {0: "Febbraio", 1: {documents: 4, revenue: 22341, month: "2019/02"}}
...

```
For example if there's an invalid date, it'll return an array like this:

```
0: {0: "Gennaio", 1: {documents: 4, revenue: 15000, month: "2019/01"}}
1: {0: "Febbraio", 1: {documents: 0, revenue: 0, month: 0}}
...
```

But in this case with "data.json" there shouldn't be any invalid date.😓

**Comment by GiovanniFrigo** [comment]
logging data as a first thing inside filterData prints this: 
```0: Object { documents: 4, revenue: 15000, month: "2019/01" }
1: Object { documents: 4, revenue: 22341, month: "2019/02" }
2: Object { documents: 5, revenue: 32000, month: "2019/03" }
3: Object { documents: 1, revenue: 900, month: "2019/04" }
4: Object { documents: 2, revenue: 29000, month: "2019/05" }
5: Object { documents: 1, revenue: 28000, month: "2019/06" }
6: Object { documents: 2, revenue: 28000, month: "2019/07" }
7: Object { documents: 1, revenue: 12000, month: "2019/08" }
8: Object { documents: 0, revenue: 0, month: "2019/09" }
9: Object { documents: 3, revenue: 32000, month: "2019/010" }
10: Object { documents: 3, revenue: 32000, month: "2019/11" }
11: Object { documents: 2, revenue: 28000, month: "2019/12" }
```

which shows the data has been loading correctly. 
When printing the  `mesi` object before the return at the end of filterData, however, I get this: 
```
[
  ["Gennaio",{"documents":0,"revenue":0,"month":0}],
  ["Febbraio",{"documents":0,"revenue":0,"month":0}],
  ["Marzo",{"documents":0,"revenue":0,"month":0}],
  ["Aprile",{"documents":0,"revenue":0,"month":0}],
  ["Maggio",{"documents":0,"revenue":0,"month":0}],
  ["Giugno",{"documents":0,"revenue":0,"month":0}],
  ["Luglio",{"documents":0,"revenue":0,"month":0}],
  ["Agosto",{"documents":0,"revenue":0,"month":0}],
  ["Settembre",{"documents":0,"revenue":0,"month":0}],
  ["Ottobre",{"documents":0,"revenue":0,"month":0}],
  ["Novembre",{"documents":0,"revenue":0,"month":0}],
  ["Dicembre",{"documents":0,"revenue":0,"month":0}]
]
```

Looks like the conversion was not successful?

**Comment by asdf1899** [comment]
Yes it looks like that in filterData() there's something wrong. Maybe a solution could be fixing this warning 
`(./src/components/Table.js Line 34:25: Expected to return a value in arrow function array-callback-return)` by replacing the .map() function in line 34 with this code:
```
mesi = mesi.map((mese) => {
  if (mese[0] === currentMonth){
    mese[1] = data[i]
  }
  return mese;
})
```

Another solution could be pull from **filterDataIssue** branch. In this branch I modify filterData(data) function.
I hope this solutions could fix the bug in your machines.

**Comment by GiovanniFrigo** [comment]
Unfortunately, neither of the two solutions worked for me. 

Hint: the system language of both my machines is set to english..

**Comment by asdf1899** [comment]
I've tried also on other virtual machines, but still can't find the problem and a pratical solution.

Maybe the thing in filterData() that could give problems based on the system language/timezone could be the Date instance?

```
let currentDate = new Date(data[i].month);
this.isValidDate(currentDate);
```
Line 30, 31 in Table.js

**Comment by GiovanniFrigo** [comment]
Yes, the issue is exactly that. You cannot parse a date from a non ISO-string in a free form like the one given in the data without specifying the format to be matched against, as this depends on the locale of the end user.

**Comment by asdf1899** [comment]
Ahh ok, now everything is clear. I fix it in *filterDataIssue* branch 4238799f1302b3ccb766ad224479fd54a3fc775b
Could you let me know if it works on your machine? I'm sorry for the trouble

**Comment by GiovanniFrigo** [comment]
I can confirm it's working properly now. Thanks!
